src/eko/output/struct.py

```diff
@@ -329,6 +335,8 @@ class EKO:
     """Path on disk, to which this object is linked (and for which it is
     essentially an interface).
     """
+    working_dir: pathlib.Path
```

Comment:
You can use `path`: there are not two distinct objects
- if the `EKO` is *opened*, then `path` /`wdir` is the working dir, and not the tar, since it is not associated to it (it might be loaded from a tar, as a NumPy array might be loaded from an `.npy` file, but after loading any association is lost)
- if the `EKO` is *closed*, it should not be used (on close the object should be destroyed), so again, you can dump it on a tar

So, I would decouple the two operations:
- `close()` will just clean up the folder
- `dump(path)` will save another tar (or `write_tar(path)`, here I'm not complaining about the name)

If you open with the context manager, then is the context manager that is in charge of using the same path, and it is retained in the function scope, so you don't need a class attribute for it.